### PR TITLE
Synchronize xml with KDE and Gnome

### PR DESCRIPTION
**Pull request by kurnevsky**
https://github.com/KDE/plasma-workspace/blob/master/xembed-sni-proxy/org.kde.StatusNotifierItem.xml
https://github.com/ubuntu/gnome-shell-extension-appindicator/blob/master/interfaces-xml/StatusNotifierItem.xml

### CODE REVIEW

**Comment by kurnevsky** [comment]
Though after this I get the following error:
```
status-notifier-item> src/StatusNotifier/Host/Service.hs:357:12: error:
status-notifier-item>     Not in scope: ‘I.registerForNewIconThemePath’
status-notifier-item>     Module ‘StatusNotifier.Item.Client’ does not export ‘registerForNewIconThemePath’.
status-notifier-item>     |
status-notifier-item> 357 |         , (I.registerForNewIconThemePath, handleNewIcon)
```

@IvanMalison any idea why? `<property name="IconThemePath" type="s" access="read"/>` is still there...

**Comment by kurnevsky** [comment]
Ah, I see, it's `NewIconThemePath` that's removed...

**Comment by kurnevsky** [comment]
It seems it's a valid deletion - spec doesn't mention it anymore: https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/StatusNotifierItem/

So removed this from the code as well.

**Comment by kurnevsky** [comment]
Also `NewStatus` now has an argument - commented it out for now as code should be changed somehow.

**Comment by kurnevsky** [comment]
There are also 2 new signals: `NewAttentionIcon` and `NewMenu` that should be handled by code somehow...

File: src/StatusNotifier/Host/Service.hs [hunk 1]
```diff
@@ -354,7 +354,6 @@ build Params { dbusClient = mclient
 
       clientRegistrationPairs =
         [ (I.registerForNewIcon, handleNewIcon)
-        , (I.registerForNewIconThemePath, handleNewIcon)
```
Comment (colonelpanic8) [inline src/StatusNotifier/Host/Service.hs]: Why is this removed?

**Comment by colonelpanic8** [comment]
> It seems it's a valid deletion - spec doesn't mention it anymore: https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/StatusNotifierItem/

Keep in mind that we are trying to support a few different protocols here though. KDE has its own flavor of SNI, but there is also a more gnome flavored sni that is slightly different.

Unfortunatley, it seems that its not completely standardized.

File: src/StatusNotifier/Host/Service.hs (hunk 1 above)
Comment (kurnevsky) [reply src/StatusNotifier/Host/Service.hs]: Indeed gnome handles it: https://github.com/ubuntu/gnome-shell-extension-appindicator/blob/master/interfaces-xml/StatusNotifierItem.xml
Restored.

File: src/StatusNotifier/Host/Service.hs [hunk 2]
```diff
@@ -189,6 +191,7 @@ build Params { dbusClient = mclient
                  , menuPath = menu
                  , overlayIconName = overlayIName
                  , overlayIconPixmaps = overlayPixmap
+                 , itemIsMenu = itemIsMenu
```
Comment (kurnevsky) [inline src/StatusNotifier/Host/Service.hs:194]: Should it be added here like this?

Comment (colonelpanic8) [reply src/StatusNotifier/Host/Service.hs:194]: Sure that works
